From: Sisyphus <dev@radxa.com>
Date: Thu, 27 Aug 2026 00:00:00 +0000
Subject: [PATCH] aic8800: do not compare address of bssid array to NULL

gcc 15 -Werror=address rejects:

  if (scan_re->bss && &scan_re->bss->bssid[0])

bssid is an array; its address is never NULL. Keep the pointer check
on scan_re->bss only.

Upstream-Status: Inappropriate [vendor driver]
Signed-off-by: Sisyphus <dev@radxa.com>
---
 drivers/net/wireless/aic/aic8800_usb/aic8800_fdrv/aicwf_wext_linux.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/aic/aic8800_usb/aic8800_fdrv/aicwf_wext_linux.c b/drivers/net/wireless/aic/aic8800_usb/aic8800_fdrv/aicwf_wext_linux.c
index 1111111..2222222 100644
--- a/drivers/net/wireless/aic/aic8800_usb/aic8800_fdrv/aicwf_wext_linux.c
+++ b/drivers/net/wireless/aic/aic8800_usb/aic8800_fdrv/aicwf_wext_linux.c
@@ -400,7 +400,7 @@ static char *aicwf_get_iwe_stream_mac_addr(struct rwnx_hw* rwnx_hw,
 	iwe->cmd = SIOCGIWAP;
 	iwe->u.ap_addr.sa_family = ARPHRD_ETHER;
 
-	if(scan_re->bss && &scan_re->bss->bssid[0]){
+	if (scan_re->bss) {
 	memcpy(iwe->u.ap_addr.sa_data, scan_re->bss->bssid, ETH_ALEN);
 	}
 
-- 
2.34.1
